enterprise/server/remote_execution/containers/podman/podman.go

```diff
@@ -247,17 +260,32 @@ func (c *podmanCommandContainer) Remove(ctx context.Context) error {
 	c.removed = true
 	c.mu.Unlock()
 	res := runPodman(ctx, "kill", nil /*=stdin*/, nil /*=stdout*/, "--signal=KILL", c.name)
-	return res.Error
+	if res.Error != nil {
+		return res.Error
+	}
+	if res.ExitCode == 0 || strings.Contains(string(res.Stderr), "no such container") {
+		return nil
+	}
+	return status.UnknownErrorf("podman remove failed: exit code %d, stderr: %s", res.ExitCode, string(res.Stderr))
 }
 
 func (c *podmanCommandContainer) Pause(ctx context.Context) error {
 	res := runPodman(ctx, "pause", nil /*=stdin*/, nil /*=stdout*/, c.name)
```

Comment:
nit: `if res.Error!=nil` then return the existing error? (like in Unpause)

Reply:
done